src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -59,6 +69,42 @@ def _post(self, request: Request):
         else:
             return Response("Unsupported version, we only support versions 1 to 3.", 400)
 
+    def _post_or_schedule_by_key(self, request: Request):
+        public_keys = set(request.relay_request_data.get("publicKeys") or ())
+
+        configs = {}
+        pending = []
+        for key in public_keys:
+            computed = self._get_cached_or_schedule(key)
+            if not computed:
+                pending.append(key)
+            else:
+                configs[key] = computed
+
+        if len(pending) > 0:
+            configs["pending"] = pending
+
+        return Response(configs, status=200)
+
+    def _get_cached_or_schedule(self, public_key) -> Optional[dict]:
+        """
+        Returns the config of a project if it's in the cache; else, schedules a
+        task to compute and write it into the cache.
+
+        Debouncing of the project happens after the task has been scheduled.
+        """
+        cached_config = projectconfig_cache.get(public_key)
+        if cached_config:
+            return cached_config
+
+        schedule_update_config_cache(
+            generate=True,
+            organization_id=None,
+            project_id=None,
+            public_key=public_key,
+            update_reason="project_config.post_v3",
+        )
```

Comment:
_nit_: Add a `return None` to demonstrate intent.

Reply:
+1 for adding `return None` here. It's true that it is redundant, but [explicit is better than implicit](https://peps.python.org/pep-0020/).